tests/ts_cluster_updowngrade_group_by_SUITE.erl

```diff
@@ -0,0 +1,167 @@
+             make_group_by_2_test(select_fails)
+            ],
+    Scen#scenario{tests = Tests}.
+
+make_select_grouped_field_test(DoesSelectPass) ->
+    Create = #create{ddl = "CREATE TABLE ~s ("
+                     "a SINT64 NOT NULL, "
+                     "b SINT64 NOT NULL, "
+                     "c TIMESTAMP NOT NULL, "
+                     "PRIMARY KEY ((a,b,quantum(c,1,s)), a,b,c))",
+                     expected = {ok, {[], []}}},
+
+    Insert = #insert{data = [{1,B,C} || B <- [1,2,3], C <- [1,2,3]],
+                     expected = ok},
+
+    {SelExp, AssertFn} 
+        = case DoesSelectPass of
+              select_passes -> 
+                  {{ok, {[<<"c">>], [{2},{1},{3}]}}, sorted_assert};
+              select_fails  -> 
+                  {?SELECTERROR, plain_assert}
+             end,
+    Select = #select{qry = "SELECT c FROM ~s "
+                     "WHERE a = 1 AND b = 1 AND c >= 1 AND c <= 1000 "
+                     "GROUP BY c",
+                     expected   = SelExp,
+                     assert_mod = ?MODULE,
+                     assert_fun = AssertFn},
+    
+    #test_set{testname = "grouped_field_test",
+              create  = Create,
+              insert  = Insert,
+              selects = [Select]}.
+
+make_group_by_2_test(DoesSelectPass) ->
+    Create = #create{ddl = "CREATE TABLE ~s ("
+                     "a SINT64 NOT NULL, "
+                     "b SINT64 NOT NULL, "
+                     "c TIMESTAMP NOT NULL, "
+                     "d SINT64 NOT NULL, "
+                     "e SINT64 NOT NULL, "
+                     "PRIMARY KEY ((a,b,quantum(c,1,s)), a,b,c,d))",
+                     expected   = {ok, {[], []}}},
+
+    Insert = #insert{data     = [{1,1,CE,D,CE} || CE <- lists:seq(1,1000), 
+                                                  D <- [1,2,3]],
+                     expected = ok},
+
+    {SelExp, AssertFn}
+        = case DoesSelectPass of
+              select_passes -> {{ok, {[<<"d">>, <<"AVG(e)">>], 
+                                      [{2,500.5}, {3,500.5}, {1,500.5}]}},
+                                sorted_assert};
+              select_fails  -> {?SELECTERROR, plain_assert}
+          end,
+    Select = #select{qry = "SELECT d, AVG(e) FROM ~s "
+                     "WHERE a = 1 AND b = 1 AND c >= 1 AND c <= 1000 "
+                     "GROUP BY d",
+                     expected = SelExp,
+                     assert_mod = ?MODULE,
+                     assert_fun = AssertFn},
+    
+    #test_set{testname = "group_by_2",
+              create  = Create,
+              insert  = Insert,
+              selects = [Select]}.
+ 
+sorted_assert(String, {ok, {ECols, Exp}}, {ok, {GCols, Got}}) ->
+    Exp2 = lists:sort(Exp),
+    Got2 = lists:sort(Got),
+    ts_util:assert_float(String, {ECols, Exp2}, {GCols, Got2});
+sorted_assert(String, Exp, Got) ->
+    ok = log_error(String ++ " banjo", Exp, Got),
+    fail.
+
+plain_assert(_String, Exp, Exp) ->
```

Comment:
That does seem to be desirable. One of the tests uses `avg` and thus includes floats in the expected results.